flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/GlobalModVersionTest.java

```diff
@@ -1,200 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.flink.runtime.executiongraph;
-
-import org.apache.flink.api.common.JobID;
-import org.apache.flink.api.common.JobStatus;
-import org.apache.flink.runtime.concurrent.ComponentMainThreadExecutorServiceAdapter;
-import org.apache.flink.runtime.execution.ExecutionState;
-import org.apache.flink.runtime.executiongraph.failover.FailoverStrategy;
-import org.apache.flink.runtime.executiongraph.failover.FailoverStrategy.Factory;
-import org.apache.flink.runtime.executiongraph.restart.InfiniteDelayRestartStrategy;
-import org.apache.flink.runtime.executiongraph.utils.SimpleSlotProvider;
-import org.apache.flink.runtime.jobgraph.JobGraph;
-import org.apache.flink.runtime.jobgraph.JobVertex;
-import org.apache.flink.runtime.testtasks.NoOpInvokable;
-import org.apache.flink.util.TestLogger;
-
-import org.junit.Test;
-
-import java.util.Random;
-
-import static org.apache.flink.runtime.executiongraph.ExecutionGraphTestUtils.waitUntilExecutionState;
-import static org.junit.Assert.assertEquals;
-import static org.mockito.Mockito.any;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-
-public class GlobalModVersionTest extends TestLogger {
```

Comment:
This test relies on the legacy failGlobal process and is outdated.

Reply:
I am not sure these tests are not relevant for the new scheduling, like concurrent global/local failure attempts

Reply:
`globalModeVersion` does not make any difference when using `DefaultScheduler` so it is not needed anymore.
Concurrent global/local failure is handled in `DefaultScheduler` with `ExecutionVertexVersioner` and its correctness is tested in `DefaultSchedulerTest`, i.e. `skipDeploymentIfVertexVersionOutdated()`, `releaseSlotIfVertexVersionOutdated` and `vertexIsNotAffectedByOutdatedDeployment`.

Reply:
True, we test the new versioning in `DefaultSchedulerTest` and `globalModeVersion` is not used any more.
We still have global/local failures in `DefaultScheduler` though. My question was more about whether we need the tests for global/local/double failures in `DefaultSchedulerTest` to see how this affects job state/versions, similar to what we have in e.g. `GlobalModVersionTest::testNoLocalFailoverWhileFailing `.

Reply:
global/local/double failures are tested in `DefaultSchedulerTest ` via `skipDeploymentIfVertexVersionOutdated`.